server/types/any.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"github.com/lib/pq/oid"
+)
+
+// Any is a type that may contain any type.
+var Any = DoltgresType{
+	OID:           uint32(oid.T_any),
+	Name:          "any",
+	Schema:        "pg_catalog",
+	Owner:         "doltgres", // TODO
```

Comment:
I've got the privilege system in, so you can add the owner to that instead of storing it in the type. You can do so with something like:
```go
postgresRole := GetRole("postgres")
AddOwner(OwnershipKey{
	PrivilegeObject: PrivilegeObject_TYPE,
	Schema:          "pg_catalog",
	Name:            "any",
}, postgresRole.ID())
```
This is all in the `server/auth` package. You could even add this to a new `typesInitDefault` function in `server/auth/database.go` under `dbInitDefault`, and call it from within `dbInitDefault` so that it's always set whenever the auth file is created. Then we can remove this from `DoltgresType` altogether.